### ISSUE TITLE
Axios removes false values from headers

### ISSUE THREAD

**Issue by apoorva-saxena** [axios/axios#6013]
### Describe the bug

I am sending these values in the header:
Accept: 'application/json, text/plain, */*',
      cookie: 'cookies',
      'x-client': 'mobile_web',
      'x-currency': 'GBP',
      'x-locale': 'en-GB',
      'x-market': 'UK',
      is_optedin_for_personalised: false,

The header is_optedin_for_personalised is being removed from the header as it is false.

Can the header stay as false and be sent to the backend service

**Comment by apoorva-saxena** [comment]
https://stackoverflow.com/questions/74994821/axios-ignores-false-value-in-request-header-while-doing-put-operation/77323920?noredirect=1#comment136317045_77323920

**Comment by DigitalBrainJS** [comment]
No. `false` and `null` are special values that are intended to mark the header as "ignored". They wouldn't be stringified.

**Comment by Chhunneng** [comment]
Did you try to use AxiosRequestConfig interface to set the headers property to a new object that includes the header. 

For example:
```js
const axios = require('axios');

const config = {
  headers: {
    Accept: 'application/json, text/plain, /',
    cookie: 'cookies',
    'x-client': 'mobile_web',
    'x-currency': 'GBP',
    'x-locale': 'en-GB',
    'x-market': 'UK',
    is_optedin_for_personalised: false,
  },
};

axios.get('https://api.example.com', config);
```

**Comment by rjewsbfd** [comment]
@WillianAgostini  thanks for the info. That is pretty silly but whatever i guess.

**Comment by robross0606** [comment]
Assuming `null` and `false` are the same thing in headers and summarily removing them without an option to keep them is pretty gnarly.  It is usually safer to be explicit about what we're doing.  Assuming behavior by way of something being absent is dangerous.  We need to explicitly set a header to `true` or `false`.  We don't want the absence of that header to be the same as `false`.  Amongst other things, that could be a security risk.  Someone doesn't even have to know the header to send in order to indicate `false` for that header on the receiving side.  What about adding an option to keep them with default behavior being as-is today?

**Comment by jasonsaayman** [comment]
closing but i will put this on my list of things to potentially look into in the future
